140-runner-host-mounts/proposal.md

```diff
@@ -0,0 +1,76 @@
+# Summary
+
+This proposal introduces a mechanism through which tasks can request files, directories, or
+block devices to be mounted from the host the worker is on to the task's container.
+
+The worker must be configured to allow tasks to mount a given location, which we will make clear
+is a last-resort option when other mechanisms are unsuitable.
+
+# Motivation
+
+When performing tasks that require access to a device on the host, such as a GPU, it is required to
+make the 'files' that the kernel creates available in a task's container.
+
+For instance, to use an AMD GPU within a container, the container needs access to these paths from the host:
+
+  - `/dev/kfd` - GPU compute interface
+  - `/dev/dri` - folder containing an interface for each GPU.
+
+Currently, it is not possible to run these workloads using Concourse.
+
+# Proposal
+
+## Worker configuration option `--allowed-host-mounts`
+
+Add a new configuration option to the worker command, `--allowed-host-mounts`:
+The value should be a regular expression. Any host path that the expression fully matches will be *allowed* to be
+mounted into a task's container if it is requested.
+
+This option can be specified only once, but can use `|` to match separate paths as required.
+
+## Task configuration option `host_mounts`
+
+Add a new configuration key (`host_mounts`) to the `task-config` schema, which can be used to request paths from the host to be
```

Comment:
I'm debating if we should make `host_mounts` in the `task-config` or the `task-step` schema.

`task-config`
```yaml
- task: my-task
  config:
    platform: linux
    ...
    host_mounts:
      - /host/path:/container/path
```

or `task-step`
```yaml
- task: my-task
  host_mounts:
    - /host/path:/container/path
  config:
    platform: linux
    ...
```

I'm debating this because this config option feels similar to the `task.privileged` option, which is only configurable in the `task-step` schema. I know `privileged` is only configurable there because we wanted to make it easy for folks to see where in their pipelines "risky" stuff was happening.

I'm leaning towards putting this in the `task-step` schema. Wdyt?

Reply:
I guess it makes sense to put it in the same place as privileged